roles/openshift_certificate_expiry/library/openshift_cert_expiry.py

```diff
@@ -129,7 +131,7 @@ def _parse_cert(self):
                 not_after_raw = l.partition(' : ')[-1]
                 # Last item: ('Not After', ' : ', 'Feb  7 18:19:35 2019 GMT')
                 not_after_parsed = datetime.datetime.strptime(not_after_raw, '%b %d %H:%M:%S %Y %Z')
-                self.not_after = not_after_parsed.strftime('%Y%m%d%H%M%SZ')
+                self.not_after = not_after_parsed.strftime('%Y%m%d%H%M%SZ').encode()
```

Comment:
What is intended here?

In both Py2/Py3 `datetime.strftime` returns a `str`, though those types are not the same in the two language versions.
Calling `.encode` will "reencode" in Py2 and return a `str`, while in Py3 it will return a b"byte string".

Since the default encodings are different, I'd suggest we use an explicit encoding here.


---

I didn't stop to understand the big picture, but it is strange that we're storing a string representation of the time... would be more natural to parse the file and store the `datetime.datetime` object, and then serialize when writing.